### PR TITLE
docs: Update the version matrix

### PR DESCRIPTION
**Pull request by sayboras**
This is not only to include the recent releases, but also to correct envoy versions for old releases.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,15 +15,21 @@ Other combinations may work but are not tested.
 | Cilium Version | Envoy version |
 |----------------|---------------|
 | (main)         | v1.28.x       |
+| v1.15.2        | v1.27.3       |
 | v1.15.1        | v1.27.3       |
 | v1.15.0        | v1.27.2       |
-| v1.14.5        | v1.26.7       |
+| v1.14.8        | v1.26.7       |
```
Comment (mhofstetter) [inline README.md:21]: i miss `v1.14.7` :thinking:

Comment (sayboras) [reply README.md:21]: ah good catch :+1:

**Review by mhofstetter — APPROVED** [review]
Thanks :ship:

**Comment by sayboras** [comment]
Merge this in as this is only docs change.

**Pull request merged**